docs/source/llms/rag/notebooks/mlflow-e2e-evaluation.ipynb

```diff
@@ -0,0 +1,918 @@
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "d2c6bff7-d988-4f09-ac10-6c9ea14b9242",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def evaluate_chunk_size(chunk_size):\n",
+    "    list_of_documents = loader.load()\n",
+    "    text_splitter = CharacterTextSplitter(chunk_size=chunk_size, chunk_overlap=0)\n",
+    "    docs = text_splitter.split_documents(list_of_documents)\n",
+    "    embedding_function = SentenceTransformerEmbeddings(model_name=\"all-MiniLM-L6-v2\")\n",
+    "    retriever = Chroma.from_documents(docs, embedding_function).as_retriever()\n",
+    "\n",
+    "    def retrieve_doc_ids(question: str) -> List[str]:\n",
+    "        docs = retriever.get_relevant_documents(question)\n",
+    "        doc_ids = [doc.metadata[\"source\"] for doc in docs]\n",
+    "        return doc_ids\n",
+    "\n",
+    "    def retriever_model_function(question_df: pd.DataFrame) -> pd.Series:\n",
+    "        return question_df[\"question\"].apply(retrieve_doc_ids)\n",
+    "\n",
+    "    with mlflow.start_run() as run:\n",
+    "        evaluate_results = mlflow.evaluate(\n",
+    "            model=retriever_model_function,\n",
+    "            data=eval_data,\n",
+    "            model_type=\"retriever\",\n",
+    "            targets=\"source\",\n",
+    "            evaluators=\"default\",\n",
+    "        )\n",
+    "    return evaluate_results\n",
+    "\n",
+    "\n",
+    "result1 = evaluate_chunk_size(1000)\n",
+    "result2 = evaluate_chunk_size(2000)\n",
+    "\n",
+    "\n",
+    "display(result1.tables[\"eval_results_table\"])\n",
+    "display(result2.tables[\"eval_results_table\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "bd45cf0e-e139-4059-a2bd-6e4fc4d5d36e",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## Evaluate the RAG system using `mlflow.evaluate()`\n",
+    "Create a simple function that runs each input through the RAG chain"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {
+      "byteLimit": 2048000,
+      "rowLimit": 10000
+     },
+     "inputWidgets": {},
+     "nuid": "667ec809-2bb5-4170-9937-6804386b41ec",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def model(input_df):\n",
+    "    return input_df[\"questions\"].map(qa).tolist()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "application/vnd.databricks.v1+cell": {
+     "cellMetadata": {},
+     "inputWidgets": {},
+     "nuid": "d1064306-b7f3-4b3e-825c-4353d808f21d",
+     "showTitle": false,
+     "title": ""
+    }
+   },
+   "source": [
+    "## Create an eval dataset (Golden Dataset)"
```

Comment:
It would be helpful to have some description on what we evaluate on this dataset (relevance + latency), and why questions only are sufficient for these metrics.

Reply:
Agreed. I'll add some context to these main sections just so there's something to follow along with.